tests/foreman/cli/test_satellitesync.py

```diff
@@ -415,58 +635,92 @@ def test_positive_export_import_default_org_view(
-                'product-id': importing_prod['id'],
+                'name': cv_name,
+                'organization-id': export_org['id'],
             }
         )
-        ContentView.version_import(
-            {'export-tar': exported_tar, 'organization-id': importing_org['id']}
+        # Verify packages
+        default_cvv_id = content_view['versions'][0]['id']
+        cv_packages = Package.list({'content-view-version-id': default_cvv_id})
+        assert len(cv_packages)
+        # Verify export directory is empty
+        assert validate_filepath(default_sat, export_cv_dir) == ''
+        # Export complete library
+        path = ContentExport.completeLibrary({'organization-id': export_org['id']})
+        # grab export dir and check all exported files are there
+        export_dir = os.path.dirname(path['message'])
+        export_folder = os.path.split(export_dir)
+        result = default_sat.execute(f'ls {export_dir}')
+        assert result.stdout != ''
+        # Verify 'export-library' is created and packages are there
+        export_lib_cv = ContentView.info(
+            {
+                'name': export_library,
+                'organization-id': export_org['id'],
+            }
         )
+        export_lib_cvv_id = export_lib_cv['versions'][0]['id']
+        exported_lib_packages = Package.list({'content-view-version-id': export_lib_cvv_id})
+        assert len(cv_packages)
+        assert exported_lib_packages == cv_packages
+        # Verify export directory is not empty
+        assert validate_filepath(default_sat, export_cv_dir) != ''
+        # importing portion
+        importing_org = make_org()
+        # set disconnected mode on and set download policy to immediate
+        Settings.set({'name': "default_download_policy", 'value': "immediate"})
+        Settings.set({'name': 'content_disconnected', 'value': "Yes"})
+        # Move export files to import location and set permission
+        default_sat.execute(f'mv {export_dir} /var/lib/pulp/imports/')
+        import_path = f'/var/lib/pulp/imports/{export_folder[1]}'
+        default_sat.execute(f'chown -R pulp:pulp {import_path}')
+        # check that files are present in import_path
+        result = default_sat.execute(f'ls {import_path}')
+        assert result.stdout != ''
+        # Import and verify content of library
+        ContentImport.library({'organization-id': importing_org['id'], 'path': import_path})
         importing_cvv = ContentView.info(
             {'name': importing_cv_name, 'organization-id': importing_org['id']}
         )['versions']
         assert len(importing_cvv) >= 1
         imported_packages = Package.list({'content-view-version-id': importing_cvv[0]['id']})
-        assert len(imported_packages) > 0
-        assert len(exported_packages) == len(imported_packages)
+        assert len(imported_packages)
+        assert len(cv_packages) == len(imported_packages)
+        # reset disconnected mode and download policy
```

Comment:
This `reset` action needs to be done in a finalizer, set immediately after the setting is modified.

There are several actions/assertions in the test that could cause an exception - meaning the setting is never restored. This causes very difficult to debug test failures, because a failure in this test case impacts the satellite instance for the remainder of the test session.

I would recommend using the decorator implementation of registering a finalizer, since its more than one statement.

```
@request.addfinalizer
def _settings_reset():
    Settings.set({'name': "default_download_policy", 'value': "on_demand"})
    Settings.set({'name': 'content_disconnected', 'value': "No"})
```

Furthermore, it would be best to read the original setting, instead of assuming and hard coding a value that's the default at the time of writing the test.